### PR TITLE
Refactor menus/dropdowns and duplicate button

### PR DESCRIPTION
**Pull request by Raenore**
This is a chonky one and the diff might be quite something. Check the changes per commit if that's easier for you.

### This is quite big in scope, but it has the following uses:
- Port over all dropdowns using `displayDropDown()` to the new `TRP3_MenuUtil`.
- Create some consistency for the "More options" dropdowns.
- To mimic Blizzard, we use onMouseDown instead of onClick to access these dropdowns.
- Toolbar and TargetFrame buttons can now have an `onMouseDown` property.

### This fixes a few things:
- Additive/Creation lines in dropdowns are colored in green.
- Destructive/Remove lines in dropdowns are colored in red.
- Right-click on RP status toolbar (IC/OOC) now has the profile list in alphabetical order.

### Other things:
- The icon copy dropdown buttons have been simplified a bit (thanks Meorawr).

> [!NOTE]
> I am not entirely happy about `onGlanceSlotClick` dropdown rewrite due to how `getGlanceMenuEntries` is set up.
>
> It is obviously still written for how the old dropdown table would receive its info. However currently all the `displayDropDown()` has been removed with all the output being the same or improved where possible.
>
> Refactoring further underlying code should happen at some point, so that snippets like that don't have to exist.

> [!IMPORTANT]
> Three new localization keys:
> - `CM_OPTIONS`
> - `CM_OPTIONS_ADDITIONAL`
> - `CM_DUPLICATE`

## Add additional info + Personality traits duplicate button
https://github.com/Total-RP/Total-RP-3/assets/172234435/c8e819d4-01f4-4a7c-9634-23bb8cc82a4f

This ports over the "Remove" button that both of these options had into a dropdown button.

This dropdown has two options:
- Duplicate
- Delete

Duplicate will make a copy of the current additional info or personality trait line (including custom colors if set), adding it at the bottom of the list.
Delete has been moved lower, and made red, as it is a destructive option. This will hopefully make it much harder to click by accident.

Finally the dropdown button for additional information has been moved a bit to the right so it is on the same horizontal location as the personality trait dropdown.

### CODE REVIEW

File: totalRP3/Modules/ModuleManagement.lua [hunk 1]
```diff
@@ -456,8 +456,8 @@ function onModuleStarted()
 		previous = frame;
 		frame:Init(module);
 		local actionButton = frame.Action;
-		setTooltipAll(actionButton, "BOTTOMLEFT", 10, 10, loc.CM_ACTIONS);
-		actionButton:SetScript("OnClick", onActionClicked);
+		setTooltipAll(actionButton, "TOP", 0, 5, loc.CM_OPTIONS, TRP3_API.FormatShortcutWithInstruction("CLICK", loc.CM_OPTIONS_ADDITIONAL));
+		actionButton:SetScript("onMouseDown", onActionClicked);
```
Comment (Meorawr) [inline totalRP3/Modules/ModuleManagement.lua:460]:
```suggestion
		actionButton:SetScript("OnMouseDown", onActionClicked);
```

File: totalRP3/Core/Profiles.lua [hunk 2]
```diff
@@ -496,13 +495,11 @@ function TRP3_API.profile.init()
 
 			end
 		end);
-		_G[widget:GetName().."Action"]:SetScript("OnClick", onActionClicked);
+		_G[widget:GetName().."Action"]:SetScript("onMouseDown", onActionClicked);
```
Comment (Meorawr) [inline totalRP3/Core/Profiles.lua:498]:
```suggestion
		_G[widget:GetName().."Action"]:SetScript("OnMouseDown", onActionClicked);
```

File: totalRP3/Core/Popup.lua [hunk 3]
```diff
@@ -1014,7 +1005,7 @@ local function initColorBrowser()
 		end
 	end);
 
-	TRP3_ColorBrowserPresets:SetScript("OnClick", colorPresetsDropDown);
+	TRP3_ColorBrowserPresets:SetScript("onMouseDown", colorPresetsDropDown);
```
Comment (Meorawr) [inline totalRP3/Core/Popup.lua:1008]:
```suggestion
	TRP3_ColorBrowserPresets:SetScript("OnMouseDown", colorPresetsDropDown);
```

File: totalRP3/Modules/Register/Characters/RegisterAbout.lua [hunk 4]
```diff
@@ -979,13 +978,13 @@ function TRP3_API.register.inits.aboutInit()
...
-				description:CreateButton(loc.UI_ICON_COPY, function() TRP3_API.SetLastCopiedIcon(icon); end);
-				description:CreateButton(loc.UI_ICON_COPYNAME, function() TRP3_API.popup.showCopyDropdownPopup({icon}); end);
+				description:CreateButton(loc.UI_ICON_COPY, TRP3_API.SetLastCopiedIcon, icon);
+				description:CreateButton(loc.UI_ICON_COPYNAME, TRP3_API.popup.showCopyDropdownPopup, {icon});
 				description:CreateButton(loc.UI_ICON_PASTE, function() onHistoIconSelected(TRP3_API.GetLastCopiedIcon()); end);
 			end);
 		end
 	end);
-	TRP3_RegisterAbout_Edit_Music_Action:SetScript("OnClick", onMusicEditClicked);
+	TRP3_RegisterAbout_Edit_Music_Action:SetScript("onMouseDown", onMusicEditClicked);
```
Comment (Meorawr) [inline totalRP3/Modules/Register/Characters/RegisterAbout.lua:987]:
```suggestion
	TRP3_RegisterAbout_Edit_Music_Action:SetScript("OnMouseDown", onMusicEditClicked);
```

File: totalRP3/Modules/Register/Characters/RegisterCharacteristics.lua [hunk 5]
```diff
@@ -1119,8 +1138,13 @@ function setEditDisplay()
 			frame = CreateFrame("Frame", "TRP3_RegisterCharact_MiscEditLine" .. frameIndex, TRP3_RegisterCharact_Edit_CharactPanel_Container, "TRP3_RegisterCharact_MiscEditLine");
 			_G[frame:GetName() .. "NameFieldText"]:SetText(loc.CM_NAME);
 			_G[frame:GetName() .. "ValueFieldText"]:SetText(loc.CM_VALUE);
-			_G[frame:GetName() .. "Delete"]:SetScript("OnClick", onMiscDelete);
-			setTooltipForSameFrame(_G[frame:GetName() .. "Delete"], "TOP", 0, 5, loc.CM_REMOVE);
+			_G[frame:GetName() .. "Action"]:SetScript("onMouseDown", function(self)
```
Comment (Meorawr) [inline totalRP3/Modules/Register/Characters/RegisterCharacteristics.lua:1141]:
```suggestion
			_G[frame:GetName() .. "Action"]:SetScript("OnMouseDown", function(self)
```

File: totalRP3/Modules/Register/Characters/RegisterCharacteristics.lua [hunk 6]
```diff
@@ -1186,7 +1210,13 @@ function setEditDisplay()
 		if frame == nil then
 			-- Create psycho attribute widget if not already exists
 			frame = CreateFrame("Frame", "TRP3_RegisterCharact_PsychoEditLine" .. frameIndex, TRP3_RegisterCharact_Edit_CharactPanel_Container, "TRP3_RegisterCharact_PsychoInfoEditLine");
-			frame.DeleteButton:SetScript("OnClick", onPsychoDelete);
+			frame.ActionButton:SetScript("onMouseDown", function(self)
```
Comment (Meorawr) [inline totalRP3/Modules/Register/Characters/RegisterCharacteristics.lua:1213]:
```suggestion
			frame.ActionButton:SetScript("OnMouseDown", function(self)
```

File: totalRP3/Modules/Register/Characters/RegisterCharacteristics.lua [hunk 7]
```diff
@@ -1573,16 +1601,16 @@ function TRP3_API.register.inits.characteristicsInit()
...
+				description:CreateButton(loc.UI_ICON_COPYNAME, TRP3_API.popup.showCopyDropdownPopup, {icon});
 				description:CreateButton(loc.UI_ICON_PASTE, function() onPlayerIconSelected(TRP3_API.GetLastCopiedIcon()); end);
 			end);
 		end
 	end);
 	TRP3_RegisterCharact_NamePanel_Edit_CancelButton:SetScript("OnClick", showCharacteristicsTab);
 	TRP3_RegisterCharact_NamePanel_Edit_SaveButton:SetScript("OnClick", onSave);
 	TRP3_RegisterCharact_NamePanel_EditButton:SetScript("OnClick", onEdit);
-	TRP3_RegisterCharact_ActionButton:SetScript("OnClick", onActionClicked);
+	TRP3_RegisterCharact_ActionButton:SetScript("onMouseDown", onActionClicked);
```
Comment (Meorawr) [inline totalRP3/Modules/Register/Characters/RegisterCharacteristics.lua:1613]:
```suggestion
	TRP3_RegisterCharact_ActionButton:SetScript("OnMouseDown", onActionClicked);
```

File: totalRP3/Modules/Register/Characters/RegisterCharacteristics.lua [hunk 8]
```diff
@@ -1608,7 +1636,18 @@ function TRP3_API.register.inits.characteristicsInit()
 	TRP3_RegisterCharact_Edit_ClassButton.onSelection = onClassColorSelected;
 	TRP3_RegisterCharact_Edit_EyeButton.onSelection = onEyeColorSelected;
 
-	setupDropDownMenu(TRP3_RegisterCharact_Edit_PsychoAdd, PSYCHO_PRESETS_DROPDOWN, psychoAdd, 0, true, false);
+	TRP3_RegisterCharact_Edit_PsychoAdd:SetScript("onMouseDown", function(button)
```
Comment (Meorawr) [inline totalRP3/Modules/Register/Characters/RegisterCharacteristics.lua:1639]:
```suggestion
	TRP3_RegisterCharact_Edit_PsychoAdd:SetScript("OnMouseDown", function(button)
```

File: totalRP3/Modules/Register/Companions/RegisterCompanionsPage.lua [hunk 9]
```diff
@@ -437,8 +435,8 @@ TRP3_API.RegisterCallback(TRP3_Addon, TRP3_Addon.Events.WORKFLOW_ON_LOAD, functi
 	TRP3_CompanionsPageInformationConsult_About_Empty:SetText(loc.REG_PLAYER_ABOUT_EMPTY);
 	TRP3_CompanionsPageInformationConsult_NamePanel_EditButton:SetText(loc.CM_EDIT);
 	setupIconButton(TRP3_CompanionsPageInformationConsult_NamePanel_ActionButton, TRP3_InterfaceIcons.Gears);
-	setTooltipForSameFrame(TRP3_CompanionsPageInformationConsult_NamePanel_ActionButton, "TOP", 0, 5, loc.CM_ACTIONS);
-	TRP3_CompanionsPageInformationConsult_NamePanel_ActionButton:SetScript("OnClick", onActionClick);
+	setTooltipForSameFrame(TRP3_CompanionsPageInformationConsult_NamePanel_ActionButton, "TOP", 0, 5, loc.CM_OPTIONS, TRP3_API.FormatShortcutWithInstruction("CLICK", loc.CM_OPTIONS_ADDITIONAL));
+	TRP3_CompanionsPageInformationConsult_NamePanel_ActionButton:SetScript("onMouseDown", onActionClick);
```
Comment (Meorawr) [inline totalRP3/Modules/Register/Companions/RegisterCompanionsPage.lua:439]:
```suggestion
	TRP3_CompanionsPageInformationConsult_NamePanel_ActionButton:SetScript("OnMouseDown", onActionClick);
```

File: totalRP3/Modules/Register/Companions/RegisterCompanionsProfiles.lua [hunk 10]
```diff
@@ -514,16 +515,14 @@ TRP3_API.RegisterCallback(TRP3_Addon, TRP3_Addon.Events.WORKFLOW_ON_LOAD, functi
 				playUISound(SOUNDKIT.IG_MAINMENU_OPTION_CHECKBOX_ON);
 			end
 		end);
-		_G[widget:GetName().."Action"]:SetScript("OnClick", onActionClicked);
+		_G[widget:GetName().."Action"]:SetScript("onMouseDown", onActionClicked);
```
Comment (Meorawr) [inline totalRP3/Modules/Register/Companions/RegisterCompanionsProfiles.lua:518]:
```suggestion
		_G[widget:GetName().."Action"]:SetScript("OnMouseDown", onActionClicked);
```

File: totalRP3/Modules/Register/Companions/RegisterCompanionsProfiles.lua [hunk 11]
```diff
@@ -514,16 +515,14 @@ TRP3_API.RegisterCallback(TRP3_Addon, TRP3_Addon.Events.WORKFLOW_ON_LOAD, functi
...
 		_G[widget:GetName().."Bound"]:SetText(loc.REG_COMPANION_BOUNDS);
 		_G[widget:GetName().."Bound"]:Show();
-		_G[widget:GetName().."Bound"]:SetScript("OnClick", onBoundClicked);
+		_G[widget:GetName().."Bound"]:SetScript("onMouseDown", onBoundClicked);
```
Comment (Meorawr) [inline totalRP3/Modules/Register/Companions/RegisterCompanionsProfiles.lua:521]:
```suggestion
		_G[widget:GetName().."Bound"]:SetScript("OnMouseDown", onBoundClicked);
```

File: totalRP3/Modules/Register/Main/RegisterList.lua [hunk 12]
```diff
@@ -992,8 +988,8 @@ TRP3_API.RegisterCallback(TRP3_Addon, TRP3_Addon.Events.WORKFLOW_ON_LOAD, functi
 	TRP3_RegisterListHeaderInfoTT:SetScript("OnClick", switchInfoSorting);
 	TRP3_RegisterListHeaderTimeTT:SetScript("OnClick", switchTimeSorting);
 
-	setTooltipForSameFrame(TRP3_RegisterListHeaderActions, "TOP", 0, 0, loc.CM_ACTIONS);
-	TRP3_RegisterListHeaderActions:SetScript("OnClick", function(self)
+	setTooltipForSameFrame(TRP3_RegisterListHeaderActions, "TOP", 0, 5, loc.CM_OPTIONS, TRP3_API.FormatShortcutWithInstruction("CLICK", loc.CM_OPTIONS_ADDITIONAL));
+	TRP3_RegisterListHeaderActions:SetScript("onMouseDown", function(self)
```
Comment (Meorawr) [inline totalRP3/Modules/Register/Main/RegisterList.lua:992]:
```suggestion
	TRP3_RegisterListHeaderActions:SetScript("OnMouseDown", function(self)
```

**Review by Meorawr — CHANGES_REQUESTED** [review]

**Comment by Raenore** [comment]
Merged in both your suggestions, thanks for that.

Going to wait for Meow's bigger review, as having the dropdowns in will help a bit with the tooltip purge (as this PR changes all the action buttons).

**Review by Meorawr — APPROVED** [review]
I'm just gonna ignore getGlanceMenuEntries.

**Review by Solanya — APPROVED** [review]

**Review by Solanya — APPROVED** [review]

**Review by Solanya — APPROVED** [review]

**Pull request merged**